src/v/config/configuration.cc

```diff
@@ -763,6 +764,15 @@ configuration::configuration()
       "Enable SASL authentication for Kafka connections.",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
       false)
+  , kafka_enable_authorization(
+      *this,
+      "kafka_enable_authorization",
+      "Enable authorization for Kafka connections."
+      "  null: Ignored; authorization is enabled with enable_sasl: true"
+      "  true: authorization is required (overrides enable_sasl)"
+      "  false: authorization is disabled (overrides enable_sasl)",
```

Comment:
Should we maybe mention that `kafka_enable_authorization == false` overrides the broker configs too `kafka_api.authentication_method`?

Reply:
Now I wonder if it should change the AuthN method. For example, sasl has a different handshake, which maybe should still go ahead, but without any restrictions.